src/mscorlib/src/System/Delegate.cs

```diff
@@ -179,7 +179,7 @@ public override int GetHashCode()
             else
                 return unchecked((int)((long)this._methodPtrAux));
             */
-            return GetType().GetHashCode();
+            return ( _target != null ? _target.GetHashCode() : 0) * 33 + GetType.GetHashCode();
```

Comment:
Also, this needs to use `RuntimeHelpers.GetHashCode` so that it is not exposed to how custom types implement GetHashCode (e.g. they can implement it as `throw new NotSupportedException()`).